src/main/java/com/gojek/beast/sink/bq/BqSink.java

```diff
@@ -21,11 +27,33 @@
 
     private final BigQuery bigquery;
     private final TableId tableId;
+    private final BQResponseParser responseParser;
+    private final BQErrorHandler errorHandler; // handler instance for BQ related errors
 
     private final Stats statsClient = Stats.client();
 
     @Override
-    public InsertStatus push(Records records) {
+    public Status push(Records records) {
+        InsertAllResponse response = insertIntoBQ(records);
+        InsertStatus status = new InsertStatus(!response.hasErrors(), response.getInsertErrors());
+        //if bq has errors
+        if (response.hasErrors()) {
+            //parse the error records
+            Map<Record, List<BQInsertionRecordsErrorType>> parsedRecords = responseParser.parseBQResponse(records, response);
+            //sink error records
+            boolean isTheSinkSuccessful = errorHandler.handleErrorRecords(parsedRecords).isSuccess();
```

Comment:
could we time this, as GCS push could take long time, and could result in lag at time, it should be evident which ones's responsible

Reply:
it's timed in the GCS writer and the `insertToBQ` in BqSink times the bq insertion.